src/global_distrib/mod_global_distrib_utils.erl

```diff
@@ -244,12 +250,25 @@ translate_opt(Opt) ->
 -spec to_ip_tuples(Addr :: inet:ip_address() | string()) ->
                          {ok, [inet:ip_address()]} | {error, {V6 :: atom(), V4 :: atom()}}.
 to_ip_tuples(Addr) ->
-    case inet:getaddrs(Addr, inet6) of
-        {ok, Av6s} -> {ok, Av6s};
-        {error, Reasonv6} ->
-            case inet:getaddrs(Addr, inet) of
-                {ok, Av4s} -> {ok, Av4s};
-                {error, Reasonv4} -> {error, {Reasonv6, Reasonv4}}
-            end
+    case {inet:getaddrs(Addr, inet6), inet:getaddrs(Addr, inet)} of
+        {{error, Reason6}, {error, Reason4}} ->
+            {error, {Reason6, Reason4}};
+        {Addrs, {error, _}} ->
```

Comment:
And that's how it was done before as well. I have not changed this way of handling errors here. I suspect that we simply want all working IPs and only return error when both fail (because we lack connection now). 

We don't print error further on so as long as there are any IPs it is fine - we can connect. But as I said it is not even my solution, it was here before, this commit is not modifying it.

If this should be changed - I think it is a seperate issue. We would need to think about a certain way of informing user that errors appeared when fetching one version of IP address and still give him a connection with IPs from the other version.

I hope that makes sense, @arcusfelis :)

Reply:
resolved

Reply:
@kanes115 Actually, the original version did not fetch "all working IPs":

Reply:
I propose to at least add a debug log for this case.